### ISSUE TITLE
invalid compile-time error in CFE: method name hides class name in extends clause

### ISSUE THREAD

**Issue by sigmundch** [dart-lang/sdk#34870]
This program used to be compiled without a problem:

```dart
class A extends B {
  B() {}
}

class B {
}

main() => new A();
```

but it now produces this compile-time error:
```
a.dart:1:17: Error: 'B' isn't a type.
class A extends B {
                ^
```

**Comment by eernstg** [comment]
Just checking, the specification does handle this case today as it should:
```
The scope of the \EXTENDS{} and \WITH{} clauses of a class $C$
is the type-parameter scope of $C$.
...
The scope of the \IMPLEMENTS{} clause of a class $C$
is the type-parameter scope of $C$.
```
This means that the method named `B` is not in scope when we check `extends B`, because instance and static methods are in a scope which is strictly more deeply nested. Hence, I'll add label 'type-bug' to this issue.

Note that the error is reported by Dart VM version: 2.1.0-edge.02651515eb1f8cb38ad8b8deb5d01bb7c256f17b, but Dart VM version: 2.1.0-dev.7.1 correctly accepts the example, so it's due to a rather recent change.

**Comment by eernstg** [comment]
(github is a bit erratic today, it did not show Lasse's response even after reload, and I have no idea why it removed 'p1-high' when I added 'type-bug' ;-)

**Comment by kmillikin** [comment]
Github is broken today.  I can't find this issue by searching for area-front-end and p1-high, either.

Proposed fix is here: https://dart-review.googlesource.com/c/sdk/+/81008

**Comment by kmillikin** [comment]
Fixed in 393fe44f8f382eff0785e9d1925c5cbea9293346.
